src/aro/Parser.zig

```diff
@@ -6443,9 +6443,12 @@ fn addExpr(p: *Parser) Error!Result {
         try rhs.expect(p);
 
         const lhs_ty = lhs.ty;
+        const rhs_ty = rhs.ty;
         if (try lhs.adjustTypes(minus.?, &rhs, p, if (plus != null) .add else .sub)) {
             if (plus != null) {
-                if (try lhs.val.add(lhs.val, rhs.val, lhs.ty, p.comp)) try p.errOverflow(plus.?, lhs);
+                if (try lhs.val.add(lhs.val, rhs.val, lhs.ty, p.comp) and
+                    rhs_ty.signedness(p.comp) != .unsigned and
+                    lhs_ty.signedness(p.comp) != .unsigned) try p.errOverflow(plus.?, lhs);
```

Comment:
`lhs.ty` is the result type of the expression after `adjustTypes` so only it needs to be checked.

Reply:
Ah, thanks, good to know! Fixed